tsdb/db.go

```diff
@@ -606,7 +607,29 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 	// to be no lower than the maxt of the last block.
 	blocks := db.Blocks()
 	minValidTime := int64(math.MinInt64)
-	if len(blocks) > 0 {
+	var samplesFound bool
+	if _, err := os.Stat(filepath.Join(dir, "wal")); os.IsExist(err) {
+		sr, err := wal.NewSegmentsReader(filepath.Join(dir, "wal"))
+		if err != nil {
+			panic(err)
+		}
+		reader := wal.NewReader(sr)
+		dec := record.Decoder{}
+		for reader.Next() {
+			currRecord := reader.Record()
+			if record.Samples == dec.Type(currRecord) {
+				var samples []record.RefSample
+				samples, err := dec.Samples(currRecord, samples)
+				if err != nil {
+					return nil, errors.Wrap(err, "samples append")
+				}
+				minValidTime = samples[0].T
```

Comment:
The first sample is not required to be the one with the lowest timestamp.

Reply:
I think I did this since it was written in the issue as a desirable approach. Well, then should it be the minimum timestamp?

Reply:
Yes, we need to consider all the samples.